src/components/viewer/Viewer.tsx

```diff
@@ -115,10 +121,12 @@ export default class Viewer extends React.Component<Partial<IViewerProps>, Parti
       const attrs = annotation.attributes;
       return (
         <ViewerItem
-          key={annotation.id}
-          annotation={annotation}
-          onDelete={this.onItemDelete}
-          onEdit={this.onItemEdit}
+            key={annotation.id}
```

Comment:
the key solution/"hack" is kinda OK - it has one downside however. Whenever the `annotation.id` has changed, the `ViewerItem` component/object is dropped and a new one is created.

If we have multiple annotations next to each other this can potentially introduce memory bloat.

I'd say the method `componentDidUpdate` is fine to reset timer

Reply:
Nvm, I haven't noticed that it's generated from an array

Reply:
Just to document our Slack discussion:
We're not sure what is the chance that the timer set with previous props/annotation is going to fire before it is cleared in `componentDidUpdate`.

According to [this post from StackOverflow](https://stackoverflow.com/questions/33613728/what-happens-when-using-this-setstate-multiple-times-in-react-component) setState calls in result to setTimeout are not batched, so we can't be sure if clearing timer as late as in `componentDidUpdate` is safe.

Since we're listing annotations, we need to set `key` anyway, but it's interesting how to deal with with single child cases, when setting `key` is not necessary. It seems so far, that even in such case it is the only foolproof solution we know of.